x-pack/plugins/security_solution/public/attack_discovery/hooks/use_poll_api.tsx

```diff
@@ -138,6 +146,11 @@ export const usePollApi = ({
       if (connectorId == null || connectorId === '') {
         throw new Error('Invalid connector id');
       }
+      // edge case - clearTimeout does not always work,
+      // so we need to check if the connectorId has changed
+      if (connectorId !== connectorIdRef.current) {
```

Comment:
@andrew-goldstein this is an edge case, it's tough to get it to happen but it does happen. I'm not sure why, if you want to dig in and try to debug further go for it. If we don't catch it, a new `setTimeout` will be created for the last connector and the results from the last connector persist. The way I've seen this, the loading screen from the last connector persists to the new connector.